### PR TITLE
[esp32] add timeout for ble advertising

### PR DESCRIPTION
**Pull request by gjc13**
#### Problem
The spec 5.2.5 requires the BLE advertisements to timeout and use a
larger interval after the first 30 seconds.
Currently we advertise with a constant timeout forever.

 #### Summary of Changes

Add BLE advertisement timeout logic.

### CODE REVIEW

**Review by kkasperczyk-no — CHANGES_REQUESTED** [review]
I think that this PR is conflicting with the one pushed by me some time ago: https://github.com/project-chip/connectedhomeip/pull/5419 Generally there is an github issue covering some of your changes https://github.com/project-chip/connectedhomeip/issues/5249 and I was working on that too. The fact is that I didn't fixed the problem for all platforms, as currently platforms doesn't have common generic BLE manager layer (issue reported: https://github.com/project-chip/connectedhomeip/issues/5418), however I also introduced some valuable changes regarding BLE min and max interval and refactored setting fast and slow advertising mode.

To sum up, I think it's great that your solutions is more generic and applicable for all platforms, but I believe than we shouldn't place all functionalities to the Server.cpp file, as it's becoming messy and we should design dedicated generic layer for managing BLE. Maybe it's good idea to do this in iterative way and not block progress, nevertheless I suggest that at least we could try to work on merging changes from our PRs. What's your opinion on this?

**Comment by gjc13** [comment]
> I think that this PR is conflicting with the one pushed by me some time ago: #5419 Generally there is an github issue covering some of your changes #5249 and I was working on that too. The fact is that I didn't fixed the problem for all platforms, as currently platforms doesn't have common generic BLE manager layer (issue reported: #5418), however I also introduced some valuable changes regarding BLE min and max interval and refactored setting fast and slow advertising mode.
> 
> To sum up, I think it's great that your solutions is more generic and applicable for all platforms, but I believe than we shouldn't place all functionalities to the Server.cpp file, as it's becoming messy and we should design dedicated generic layer for managing BLE. Maybe it's good idea to do this in iterative way and not block progress, nevertheless I suggest that at least we could try to work on merging changes from our PRs. What's your opinion on this?

The PR has been updated to add timeout logic in ESP32 platform only and wew may create a separate PR to merge the BLE logic on different platforms and define a lean platform BLE interface.

**Review by kkasperczyk-no — APPROVED** [review]
> The PR has been updated to add timeout logic in ESP32 platform only and wew may create a separate PR to merge the BLE logic on different platforms and define a lean platform BLE interface.
>
Ok, I think that fixing only specific platforms is reasonable for now, as designing some common BLE layer to be well thought out may take some time.

**Review by jmartinez-silabs — APPROVED** [review]
This looks good to me! 
@kkasperczyk-no @gjc13 I read your comments. I actually also need to make modification to EFR platform for the different advertisement interval needed for TE1. From what I understand, it is ok with you that we do it on the platform side for now and make it more generic subsequently?

**Comment by kkasperczyk-no** [comment]
@jmartinez-silabs it would be perfect to have common layer and be able to create one fix applying easily to all platforms. Currently we have to handle it on our own in platform codes and even it someone wants to apply code to all platforms, it's often too hard because lack of knowledge and testing possibilities. So IMO yes, we don't have much different choice right now, than modifying each platform separately to not stop progress...

**Review by andy31415 — APPROVED** [review]

**Review by msandstedt — APPROVED** [review]

**Review by jelderton — APPROVED** [review]

**Review by jmartinez-silabs — APPROVED** [review]
